app/Libraries/Session/Store.php

```diff
@@ -30,24 +30,81 @@ class Store extends \Illuminate\Session\Store
 {
     const SESSION_ID_LENGTH = 40;
 
-    public function keyPrefix(int $userId = null)
+    /**
+     * Get the redis key prefix for the given user (excluding cache prefix).
+     *
+     * @return string
+     */
+    protected function keyPrefix(int $userId = null)
     {
         $userId = $userId ?? 'guest';
 
-        return "sessions:{$userId}:";
+        return "sessions:{$userId}";
+    }
+
+    /**
+     * Get the redis key containing the session list for the given user.
+     *
+     * @return string
+     */
+    protected function sessionListKey(int $userId = null)
+    {
+        return config('cache.prefix').':'.$this->keyPrefix($userId);
+    }
+
+    /**
+     * Get the redis key prefix of the current session (excluding cache prefix).
+     *
+     * @return string
+     */
+    protected function getCurrentKeyPrefix()
+    {
+        $sessionId = $this->getId();
+
+        return substr($sessionId, 0, strlen($sessionId) - static::SESSION_ID_LENGTH - 1);
     }
 
+    /**
+     * Destroys a session owned by the current user that is identified by the given id.
+     *
+     * @return bool
+     */
     public function destroyUserSession($sessionId)
     {
         if (Auth::check()) {
             $userId = Auth::user()->user_id;
-            $this->handler->destroy($this->keyPrefix($userId).$sessionId);
+            $fullSessionId = $this->keyPrefix($userId).":{$sessionId}";
```

Comment:
Not sure if mixing string interpolation and concatenation in single line a good thing.